pkg/mapper/iac-providers/arm/arm.go

```diff
@@ -0,0 +1,128 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package arm
+
+import (
+	"errors"
+
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/mapper/core"
+	"github.com/accurics/terrascan/pkg/mapper/iac-providers/arm/config"
+	fn "github.com/accurics/terrascan/pkg/mapper/iac-providers/arm/functions"
+	"github.com/accurics/terrascan/pkg/mapper/iac-providers/arm/types"
+	"github.com/accurics/terrascan/pkg/utils"
+)
+
+type armMapper struct{}
+
+// Mapper returns an ARM mapper for given template schema
+func Mapper() core.Mapper {
+	return armMapper{}
+}
+
+// Map transforms the provider specific template to terrascan native format.
+func (m armMapper) Map(resource interface{}, params ...map[string]interface{}) ([]output.ResourceConfig, error) {
+	r, ok := resource.(types.Resource)
+	if !ok {
+		return nil, errors.New("failed to cast resource into types.Resource")
+	}
+
+	config := output.ResourceConfig{
+		SkipRules: make([]output.SkipRule, 0),
+	}
+
+	// add skipRules if available
+	if r.Tags != nil {
+		skipRules := utils.ReadSkipRulesFromMap(r.Tags, config.ID)
+		if skipRules != nil {
+			config.SkipRules = append(config.SkipRules, skipRules...)
+		}
+	}
+
+	variables := params[0]
+	parameters := params[1]
+	config.Name = fn.LookUp(variables, parameters, r.Name).(string)
```

Comment:
this would panic if the resulting value is `nil` or not `string`.

Reply:
Changed to typ[esafe lookups